exercise-core/src/test/java/com/examples/attsw/exercise/core/controller/EmployeeControllerTest.java

```diff
@@ -0,0 +1,80 @@
+package com.examples.attsw.exercise.core.controller;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.examples.attsw.exercise.core.model.Employee;
+import com.examples.attsw.exercise.core.service.IEmployeeService;
+
+public class EmployeeControllerTest {
+
+	// SUT
+	public EmployeeController employeeController;
+	private List<Employee> listEmployees;
+	private IEmployeeService employeeService;
+
+	@Before
+	public void setUp() throws Exception {
+		employeeService = mock(IEmployeeService.class);
+		employeeController = new EmployeeController(employeeService);
+		listEmployees = new LinkedList<Employee>();
+		when(employeeService.getEmployees()).thenReturn(listEmployees);
+	}
+
+	@Test
+	public void testGetAllEmployeeWhenThereAreNoEmployee() {
+		assertGetAllEmployee("");
+	}
+
+	@Test
+	public void testGetAllEmployeeWhenThereIsOneEmployee() {
+		Employee employee = newEmployee("nameTest", "idTest");
+		listEmployees.add(employee);
+		assertGetAllEmployee(extractAllEmployeeStringFromList(listEmployees));
+	}
+	
+	@Test
+	public void testGetAllEmployeeWhenThereIsTwoEmployee() {
+		Employee employee0 = newEmployee("nameEmployee0", "idEmployee0");
+		Employee employee1 = newEmployee("nameEmployee1", "idEmployee1");
+		listEmployees.add(employee0);
+		listEmployees.add(employee1);
+		assertGetAllEmployee(extractAllEmployeeStringFromList(listEmployees));
+	}
+
+	@Test (expected = NullPointerException.class)
+	public void testGetEmployeeByIdWhenEmployeeDoesntExists() {
+		assertEquals(null, employeeController.getEmployeeById("1"));
+	}
+
+	@Test
+	public void testGetEmployeeByIdWhenEmployeeExists() {
+		Employee newEmployee = newEmployee("nameTest", "1");
+		when(employeeService.getEmployeeById("1")).thenReturn(newEmployee);
+		assertEquals(newEmployee.getName(), employeeController.getEmployeeById("1"));
+	}
+	
+	private String extractAllEmployeeStringFromList(List<Employee> allEmployee) {
```

Comment:
i don't really know if this is the right way

Reply:
it's OK.

On a side note: since we're using Java 8, such for statements should be replaced by lambdas (also in the SUT); it's easier and more readable to create such string representations with lambdas

Reply:
I'm trying to replace for with this:
`allEmployee.forEach(e -> stringAllEmployee.concat(e.toString()+System.getProperty("line.separator")));`
but it doesn't work at all, look like the concat call doesn't do nothing.

Reply:
It doesn't work because return of concat is a string and I didn't assign it to anything.

Reply:
In fact that's not the right way to do that :)